src/gwt/src/org/rstudio/core/client/widget/ModalDialogTracker.java

```diff
@@ -44,8 +48,14 @@ public static boolean isTopMost(PopupPanel panel)
    public static void onHide(PopupPanel panel)
    {
       dialogStack_.removeIf(panel::equals);
-      if (Desktop.hasDesktopFrame() && numModalsShowing() == 0)
-         DesktopMenuCallback.setMainMenuEnabled(true);
+      
+      if (Desktop.hasDesktopFrame()) {
+         int numModals = numModalsShowing();
+         if (BrowseCap.isElectron())
```

Comment:
I suppose this check is just to be safe since the menu would be disabled when a modal is displayed. That would make it impossible to show a GWT modal and an Electron modal at the same time.

Reply:
I also checked using Cmd+Shift+P to try and open another dialog but disabling the menu is probably preventing the quick access from executing the command.

Reply:
Yeah I'm not sure there are ways for the user to directly open another dialog while the menu is disabled, but I think it could be possible for a dialog to be opened:
- if a user does some things in the background which impacts the IDE state
- if the IDE has some timed background tasks that cause dialogs to open due to an error state

Your comment also makes me realize I haven't handled Electron dialogs (only GWT dialogs and dialogs based on `ModalDialog` typescript class)...I'd need to maybe wrap our calls to `dialog.<showMessage/etc>` and detect when those dialogs are closed

Reply:
Hmm I'll need to modify `src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java` so that those modals are tracked as well.

So, what I currently have in this PR only covers non-native modal dialogs. Native dialogs currently don't disable the menu bar. I just checked a QT build and native dialogs _do_ disable the menu bar.

I'll look further into this.

Reply:
I've added some handling for electron modals now. I'm not super stoked about the verbosity, but Electron doesn't have a dialog event that can be listened to (eg. show/hide events), so I've opted to create wrappers around the `dialog.<dialog function>` calls which enable/disable the menu bar.